plugins/modules/fabric_devices_workflow_manager.py

```diff
@@ -0,0 +1,4080 @@
+                if not task_id:
+                    self.msg = (
+                        "Unable to retrive the task_id for the task '{task_name}'."
+                        .format(task_name=task_name)
+                    )
+                    self.set_operation_result("failed", False, self.msg, "ERROR")
+                    return self
+
+                success_msg = (
+                    "L3 Handoff(s) with SDA Transit '{sda_l2_handoff}' added successfully in the device with IP '{ip}."
+                    .format(sda_l2_handoff=want_sda_l3_handoff, ip=device_ip)
+                )
+                self.get_task_status_from_tasks_by_id(task_id, task_name, success_msg).check_return_status()
+            except Exception as msg:
+                self.msg = (
+                    "Exception occured while adding the SDA L3 Handoff for the device '{ip}': {msg}"
+                    .format(ip=device_ip, msg=msg)
+                )
+                self.status = "failed"
+                return self
+
+            result_fabric_device_response.update({
+                "sda_l3_handoff_details": want_sda_l3_handoff
+            })
+            result_fabric_device_msg.update({
+                "sda_l3_handoff_details": "SDA L3 handoff added successfully"
+            })
+            self.msg = "Successfully added the L3 Handoff with SDA Transit."
+            self.status = "success"
+            return self
+
+        self.log(
+            "Current SDA L3 Handoff details for the device '{ip}' in Catalyst Center: {current_details}"
+            .format(ip=device_ip, current_details=have_sda_l3_handoff), "DEBUG"
+        )
+        self.log(
+            "Desired SDA L3 Handoff details for the device '{ip}' for Catalyst Center: {requested_details}"
+            .format(ip=device_ip, requested_details=want_sda_l3_handoff), "DEBUG"
+        )
+
+        # SDA L3 Handoff Exists
+        # Check if the SDA L3 Handoff required an update or not
+        if not self.requires_update(have_sda_l3_handoff,
+                                    want_sda_l3_handoff,
+                                    self.fabric_l3_handoff_sda_obj_params):
+            self.log(
+                "SDA L3 Handoff for the device '{ip}' doesn't require an update."
+                .format(ip=device_ip), "INFO"
+            )
+            result_fabric_device_msg.update({
+                "sda_l3_handoff_details": "SDA L3 Handoff doesn't require an update."
+            })
+            self.msg = "L3 Handoff with SDA Transit doesnot require an update."
+            self.status = "success"
+            return self
+
+        self.log("Updating SDA L3 Handoff for the device '{ip}'.".format(ip=device_ip), "DEBUG")
+
+        # SDA L3 Handoff requires an update or not
+        try:
+            payload = {"payload": [want_sda_l3_handoff]}
+            task_name = "update_fabric_devices_layer3_handoffs_with_sda_transit"
```

Comment:
self.log("Preparing to update SDA L3 Handoff with payload: {payload}".format(payload=payload), "DEBUG")